lib/addons/coredns/index.ts

```diff
@@ -1,15 +1,25 @@
-import { CoreAddOn } from "../core-addon";
+import { CoreAddOn, CoreAddOnProps } from "../core-addon";
+
+/**
+ * Configuration options for the coredns add-on.
+ */
+type corednsCollectorAddOnProps = CoreAddOnProps;
+
+
+const defaultProps = {
+    addOnName: 'coredns',
+    version: 'v1.9.3-eksbuild.5',
+    saName: 'coredns',
+    configurationValues: {}
+}
 
 /**
  * Implementation of CoreDns EKS add-on.
  */
 export class CoreDnsAddOn extends CoreAddOn {
 
-    constructor(version?: string) {
-        super({
-            addOnName: "coredns",
-            version: version ?? "v1.9.3-eksbuild.5",
-            saName: "coredns"
-        });
+    constructor(props?: corednsCollectorAddOnProps) {
```

Comment:
why is the variable called corednsCollector - what is the relevance of collector?
also, minor - came case for naming, so `coredns` = `coreDns`